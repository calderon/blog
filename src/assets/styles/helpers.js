import { css } from "styled-components"

const srOnly = css`
  position: absolute !important;
  width: 1px !important;
  height: 1px !important;
  padding: 0 !important;
  margin: -1px !important;
  overflow: hidden !important;
  clip: rect(0, 0, 0, 0) !important;
  white-space: nowrap !important;
  border: 0 !important;
`

const srOnlyFocusable = css`
  &:not(:focus) {
    ${srOnly};
  }
`

const skipLink = css`
  ${srOnly};
  ${srOnlyFocusable};
`

const _breakpoints = {
  xs: 480,
  sm: 640,
  md: 768,
  lg: 840,
  xl: 1024,
}

const devices = {
  xs: `(min-width: ${_breakpoints.xs}px)`,
  sm: `(min-width: ${_breakpoints.sm}px)`,
  md: `(min-width: ${_breakpoints.md}px)`,
  lg: `(min-width: ${_breakpoints.lg}px)`,
  xl: `(min-width: ${_breakpoints.xl}px)`,
}

export { srOnly, srOnlyFocusable, skipLink, devices }
