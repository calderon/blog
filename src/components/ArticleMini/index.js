import React from "react"
import styled from "styled-components"
import { rgba } from "polished"

import { devices } from "../../assets/styles/helpers"

import { Link } from "../Link"

const StyledArticleHeader = styled.header`
  figure,
  img {
    width: 100%;
  }

  h1 {
    margin: 1.4rem 2.5rem 1.7rem;
    font-weight: 500;
    font-size: 2rem;

    @media ${devices.lg} {
      font-size: 1.8rem;
    }
  }
`

const ArticleHeader = ({ article }) => {
  return (
    <StyledArticleHeader>
      <figure>
        <img src="https://via.placeholder.com/500x200" />
      </figure>

      <h1>{article.title}</h1>
    </StyledArticleHeader>
  )
}

const StyledArticleContent = styled.div`
  margin: 1.4rem 2.5rem;
  font-size: 1.6rem;
  font-weight: 400;

  @media ${devices.lg} {
    font-size: 1.4rem;
  }
`

const ArticleContent = ({ article }) => {
  return (
    <StyledArticleContent
      dangerouslySetInnerHTML={{ __html: article.summary }}
    />
  )
}

const StyledArticleMini = styled.article`
  transition: background 600ms, box-shadow 300ms;
  background: rgba(0, 0, 0, 0.04);
  color: ${props => props.theme.palette.primary};
  position: relative;

  &:hover {
    box-shadow: 0 0.3rem 1rem 0 rgba(0, 0, 0, 0.1);
    background: ${props => props.theme.colors.bright};
    cursor: pointer;
  }
`

const ArticleLink = styled(Link)`
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
`

const ArticleMini = props => {
  const { article, ...others } = props

  return (
    <StyledArticleMini {...others}>
      <ArticleHeader article={article.frontmatter} />
      <ArticleContent article={article.frontmatter} />
      <ArticleLink to={article.fields.slug}>
        <span className="srOnly">Go to article</span>
      </ArticleLink>
    </StyledArticleMini>
  )
}

export default ArticleMini
