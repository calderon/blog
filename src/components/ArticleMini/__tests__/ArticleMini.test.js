import React from "react"
import { render } from "../../../../__tests__/helpers/render"

import ArticleMini from "../"

const renderComponent = ({ article = {}, ...props } = {}) => render(<ArticleMini article={article} {...props} />)

describe("ArticleMini", () => {
  const article = {
    frontmatter: {
      title: "",
      summary: ""
    }
  }

  it("renders correctly", () => {
    const articleComponent = renderComponent({ article })
    expect(articleComponent).toMatchSnapshot()
  })
})
