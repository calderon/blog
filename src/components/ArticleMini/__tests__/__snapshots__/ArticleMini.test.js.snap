// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleMini renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <article
        class="sc-AxheI fclTfS"
      >
        <header
          class="sc-AxhUy hwJHpZ"
        >
          <figure>
            <img
              src="https://via.placeholder.com/500x200"
            />
          </figure>
          <h1 />
        </header>
        <div
          class="sc-AxgMl jrpZCm"
        />
        <a
          class="sc-AxjAm ioxigH sc-Axmtr kZovVo"
          href=""
        >
          <span
            class="srOnly"
          >
            Go to article
          </span>
        </a>
      </article>
    </div>
  </body>,
  "container": <div>
    <article
      class="sc-AxheI fclTfS"
    >
      <header
        class="sc-AxhUy hwJHpZ"
      >
        <figure>
          <img
            src="https://via.placeholder.com/500x200"
          />
        </figure>
        <h1 />
      </header>
      <div
        class="sc-AxgMl jrpZCm"
      />
      <a
        class="sc-AxjAm ioxigH sc-Axmtr kZovVo"
        href=""
      >
        <span
          class="srOnly"
        >
          Go to article
        </span>
      </a>
    </article>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
